Assets/MixedRealityToolkit/Services/BaseDataProviderAccessCoreSystem.cs

```diff
@@ -45,21 +78,48 @@ public override void Update()
         {
             base.Update();
 
-            foreach (var provider in dataProviders)
+            try
             {
-                provider.Update();
+                foreach (IMixedRealityDataProvider provider in dataProviders)
+                {
+                    provider.Update();
+                }
             }
+            catch (InvalidOperationException) // The collection has changed, most likely due to the application changing the profile.
+            { }
```

Comment:
Why do we only accept InvalidOperationException here? Are we not logging because it'll flood the console every udpate?

Reply:
yes on the not logging question.

we do not want to mask exceptions unrelated to the collection being changed

Reply:
>we do not want to mask exceptions unrelated to the collection being changed

Isn't catching an exception and then doing nothing about it masking the exception? Should you re-throw the exception here?

If you are not re-throwing and not logging, please explain why we are ignoring this exception completely since this seems like it could be a place where a bug could get hidden very easily: an exception could be thrown, then caught, and silentingly ignored.

Reply:
I will add more detailed comments

Reply:
The following was added in in [Late]Update()

            // Applications may choose to trigger service changes (i.e. setting MixedRealityToolkit.ActiveProfile) in
            // response to user input, such as a button press. Since most code runs on Unity's main thread, the collection
            // of data providers may change mid-Update, causing an InvalidOperationException to be thrown during enumeration.
            // To avoid unneccessary error logging, we catch the exception. It is safe to abort the loop in this fashion since
            // it is extremely unlikely that the data providers being enumerated are the same as when we started enumeration.